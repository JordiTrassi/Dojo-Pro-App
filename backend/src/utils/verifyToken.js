const jwt = require('jsonwebtoken');

function verifyToken(err, user, res) {
  if (err) {
    return res.sendStatus(403);
  }

  // eslint-disable-next-line no-underscore-dangle
  const data = { _id: user._id, email: user.email };

  const token = jwt.sign(
    { user: data },
    process.env.JWT_SECRET,
    { expiresIn: process.env.TOKEN_EXPIRES_IN },
  );

  return res.json({
    token,
  });
}

module.exports = {
  verifyToken,
};
