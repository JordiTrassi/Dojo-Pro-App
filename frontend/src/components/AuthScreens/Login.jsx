/* eslint-disable no-underscore-dangle */
import React, { useState } from 'react';
import {
  SafeAreaView,
  View,
  ImageBackground,
  Image,
  Text,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import logIn from '../../redux/actions/actionLogIn';
import validateEmail from '../../utils/validateEmail';
import styles from './Login_Styles';

const image = {
  uri: 'https://www.robinju.com/wp-content/uploads/2012/10/Fushimi-Inari-011.jpg',
};
const titleimage = {
  uri: 'https://fontmeme.com/permalink/210902/8337e2e2aeed704e1256a194f328c52c.png',
};

export default function Login({ navigation }) {
  const onPressMain = () => { navigation.navigate('MainScreen'); };

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const dispatch = useDispatch();

  function handleLogIn() {
    if (validateEmail(email)) {
      dispatch(logIn(email, password));
    }
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.imageStack}>
        <ImageBackground
          source={image}
          resizeMode="contain"
          style={styles.image}
          imageStyle={styles.image_imageStyle}
        >
          <TouchableOpacity
            testID="mainScreenButton"
            onPress={onPressMain}
          >
            <Image source={titleimage} style={styles.titleimage} />
          </TouchableOpacity>
          <Text style={styles.eMail}>E-mail</Text>
          <View style={styles.rect_email}>
            <TextInput
              testID="userInputEmail"
              style={styles.input_email}
              onChangeText={(userEmail) => setEmail(userEmail)}
              value={email}
              placeholder="user@email.com"
              autoCapitalize="none"
            />
          </View>
          <Text style={styles.password}>Password</Text>
          <View style={styles.rect_password}>
            <TextInput
              testID="userInputPassword"
              style={styles.input_password}
              onChangeText={(pass) => setPassword(pass)}
              value={password}
              secureTextEntry
              placeholder="*******************"
              autoCapitalize="none"
            />
          </View>

        </ImageBackground>
        <TouchableOpacity
          testID="loginBtn"
          style={styles.button_signin}
          onPress={() => handleLogIn()}
        >
          <Text style={styles.btn_login_text}>Sign In</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}

Login.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
