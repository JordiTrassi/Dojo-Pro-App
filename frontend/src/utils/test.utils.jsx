/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

const customRender = (ui, initialState, options) => {
  const AllTheProviders = ({ children }) => (
    <Provider store={configureStore(initialState)}>
      { children }
    </Provider>
  );

  return render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
