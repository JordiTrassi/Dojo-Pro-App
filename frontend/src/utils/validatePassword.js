export default function validatePassword(password) {
  const regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/;
  return regex.test(password);
}
