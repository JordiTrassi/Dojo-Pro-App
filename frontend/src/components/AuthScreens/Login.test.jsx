/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '../../utils/test.utils';
import Login from './Login';

const navigation = {
  navigate: jest.fn(),
  push: jest.fn(),
};

const handleLogIn = jest.fn();
describe('Given a Character Component', () => {
  let screen;
  beforeEach(() => {
    screen = render(<Login navigation={navigation} />);
  });
  describe('When it is rendered', () => {
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
  describe('When email input text is changed', () => {
    test('Then the setEmail is invoked', () => {
      const emailInput = screen.getByTestId('userInputEmail');
      const setEmail = jest.fn();
      fireEvent(emailInput, 'onChangeText', setEmail);
      expect(setEmail).toHaveBeenCalled();
    });
  });
  describe('When password input text is changed', () => {
    test('Then the setPassword is invoked', () => {
      const passwordInput = screen.getByTestId('userInputPassword');
      const setPassword = jest.fn();
      fireEvent(passwordInput, 'onChangeText', setPassword);
      expect(setPassword).toHaveBeenCalled();
    });
  });
  describe('And the email and the password are valid', () => {
    beforeEach(() => {
      const logIn = jest.Mock;

      logIn.mockReturnValue({
        type: '',
      });

      const emailInput = screen.getByTestId('userInputEmail');
      const passwordInput = screen.getByTestId('userInputPassword');
      const loginButton = screen.getByTestId('loginBtn');
      fireEvent.changeText(emailInput, setEmail);
      fireEvent.changeText(passwordInput, setPassword);
      fireEvent.press(loginButton);
    });

    test('Then loginUser function should have been called', () => {
      expect(logIn).toHaveBeenCalled();
    });
  });

  describe('And you click on loginButton', () => {
    test('Then navigation.push should have been called', () => {
      const loginButton = screen.getByTestId('loginBtn');
      fireEvent.press(loginButton);
      expect(handleLogIn).toHaveBeenCalled();
    });
  });
  describe('And you click on mainScreenButton', () => {
    test('Then navigation.push should have been called', () => {
      const mainScreenButton = screen.getByTestId('mainScreenButton');
      fireEvent.press(mainScreenButton);
      expect(screen).toHaveBeenCalled();
    });
  });
});
