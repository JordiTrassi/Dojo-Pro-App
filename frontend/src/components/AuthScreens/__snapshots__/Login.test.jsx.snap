// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 761,
        "marginLeft": -201,
        "marginTop": -125,
        "width": 953,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 819,
          "left": 0,
          "position": "absolute",
          "top": 25,
          "width": 933,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "https://www.robinju.com/wp-content/uploads/2012/10/Fushimi-Inari-011.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 819,
              "width": 933,
            },
            Object {
              "opacity": 0.55,
            },
          ]
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="mainScreenButton"
      >
        <Image
          source={
            Object {
              "uri": "https://fontmeme.com/permalink/210902/8337e2e2aeed704e1256a194f328c52c.png",
            }
          }
          style={
            Object {
              "height": 50,
              "left": 225,
              "top": 200,
              "width": 350,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 26,
            "left": 257,
            "position": "absolute",
            "top": 301,
          }
        }
      >
        E-mail
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "borderColor": "#000000",
            "borderRadius": 7,
            "borderWidth": 2,
            "height": 45,
            "left": 242,
            "position": "absolute",
            "top": 337,
            "width": 300,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="user@email.com"
          style={
            Object {
              "color": "#121212",
              "fontFamily": "arial",
              "fontSize": 25,
              "marginLeft": 18,
              "marginTop": 6,
            }
          }
          testID="userInputEmail"
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 26,
            "left": 257,
            "position": "absolute",
            "top": 421,
          }
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "borderColor": "#000000",
            "borderRadius": 7,
            "borderWidth": 2,
            "height": 45,
            "marginLeft": 239,
            "marginTop": 414,
            "width": 300,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="*******************"
          secureTextEntry={true}
          style={
            Object {
              "color": "#121212",
              "fontFamily": "arial",
              "fontSize": 25,
              "marginLeft": 18,
              "marginTop": 7,
            }
          }
          testID="userInputPassword"
          value=""
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E6E6E6",
          "borderColor": "#000000",
          "borderRadius": 20,
          "borderWidth": 3,
          "elevation": 5,
          "height": 65,
          "left": 323,
          "opacity": 1,
          "position": "absolute",
          "shadowColor": "rgba(0,0,0,1)",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 0,
          "top": 780,
          "width": 135,
        }
      }
      testID="loginBtn"
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 28,
            "marginTop": 13,
            "textAlign": "center",
          }
        }
      >
        Log In
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
